setup.rst

```diff
@@ -122,22 +123,161 @@ working only on pure Python code the pydebug build provides several useful
 checks that one should not skip.
 
 
-.. _build-dependencies:
+.. _unix-compiling:
 
-Build dependencies
-------------------
+UNIX
+----
 
-The core CPython interpreter only needs a C compiler to be built; if
-you get compile errors with a C89 or C99-compliant compiler, please `open a
-bug report <https://bugs.python.org>`_.
-However, some of the extension modules will need development headers
+The core CPython interpreter only needs a C compiler to be built,
+however, some of the extension modules will need development headers
 for additional libraries (such as the ``zlib`` library for compression).
 Depending on what you intend to work on, you might need to install these
 additional requirements so that the compiled interpreter supports the
 desired features.
 
-UNIX
-----
+If you want to install these optional dependencies, consult the
+:ref:`build-dependencies` section below.
+
+If you don't need to install them, the basic steps for building Python
+for development is to configure it and then compile it.
+
+Configuration is typically:
+
+.. code-block:: bash
+
+   ./configure --with-pydebug
+
+More flags are available to ``configure``, but this is the minimum you should
+do to get a pydebug build of CPython.
+
+Once ``configure`` is done, you can then compile CPython with:
+
+.. code-block:: bash
+
+   make -s -j2
+
+This will build CPython with only warnings and errors being printed to
+stderr and utilize up to 2 CPU cores. If you are using a multi-core machine
+with more than 2 cores (or a single-core machine), you can adjust the number
+passed into the ``-j`` flag to match the number of cores you have.
+
+At the end of the built you should see a success message, possibly followed
+by a list of extension modules that haven't been built because their
+dependencies were missing:
+
+.. code-block:: none
+
+   Python build finished successfully!
+   The necessary bits to build these optional modules were not found:
+   _bz2                  _dbm                  _gdbm
+   _lzma                 _sqlite3              _ssl
+   _tkinter              _uuid                 readline
+   zlib
+   To find the necessary bits, look in setup.py in detect_modules()
+   for the module's name.
```

Comment:
I added this block to show clearly where the missing extensions are listed, and pointed out that this is expected and should not be considered an error (I've seen people being confused by this message before).